packages/core/src/external/cda/remove-b64.ts

```diff
@@ -0,0 +1,53 @@
+import { XMLParser, XMLBuilder } from "fast-xml-parser";
+import { toArray } from "@metriport/shared";
+import { PDF_MIME_TYPE } from "../../util/mime";
+
+const notesTemplateId = "2.16.840.1.113883.10.20.22.2.65";
+const b64Representation = "B64";
+
+export function removeBase64PdfEntries(payloadRaw: string): string {
+  const parser = new XMLParser({
+    ignoreAttributes: false,
+    attributeNamePrefix: "@_",
+    removeNSPrefix: true,
+  });
+  const json = parser.parse(payloadRaw);
+
+  let removedEntry = 0;
+
+  if (json.ClinicalDocument?.component?.structuredBody?.component) {
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    toArray(json.ClinicalDocument.component.structuredBody.component).forEach((comp: any) => {
+      if (
+        toArray(comp.section?.templateId).some(
+          //eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (template: any) => template?.["@_root"] === notesTemplateId
+        )
+      ) {
+        if (comp.section.entry) {
+          //eslint-disable-next-line @typescript-eslint/no-explicit-any
+          comp.section.entry = toArray(comp.section.entry).filter((entry: any) => {
+            if (
+              entry.act?.text?.["@_mediaType"] === PDF_MIME_TYPE &&
+              entry.act.text["@_representation"] === b64Representation
+            ) {
+              removedEntry++;
+              return false;
+            }
+            return true;
+          });
+        }
+      }
+    });
+  }
+
+  const builder = new XMLBuilder({
+    format: false,
+    ignoreAttributes: false,
+    attributeNamePrefix: "@_",
+    suppressEmptyNode: true,
+    suppressBooleanAttributes: false,
+  });
+  const xml = builder.build(json);
+  return removedEntry > 0 ? xml : payloadRaw;
```

Comment:
If we didnt remove anything, lets return the original payload. I am not certain that the modified payload isnt a little bit different, and I dont want to risk some sort of potential downstream converter error across all patient documents.

Reply:
Why are we building the XML w/ XMLBuilder? Is that the only way to convert the JSON representation into string? Just trying to avoid any step processing the whole of the contents.